transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/scheduling/persistence/ClientAnalyticsStore.java

```diff
@@ -0,0 +1,47 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.android.datatransport.runtime.scheduling.persistence;
+
+import com.google.android.datatransport.runtime.firebase.transport.ClientMetrics;
+import com.google.android.datatransport.runtime.firebase.transport.LogEventDropped.Reason;
+
+/**
+ * Persistent Layer
+ *
+ * <p>Responsible for storing, updating, and retrieving client analytics data.
+ */
+public interface ClientAnalyticsStore {
```

Comment:
imo the term "analytics" is very overloaded, can we use something more concrete like `ClientHealthMetricsStore` instead?

Reply:
Will change that.